eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/ImmutableBag.java

```diff
@@ -219,4 +219,15 @@
      */
     @Override
     ImmutableList<ObjectIntPair<T>> bottomOccurrences(int count);
+
+    /**
+     * Overrides toImmutableBag in RichIterable to return this.
+     *
+     * @since 11.0
+     */
+    @Override
+    default ImmutableBag<T> toImmutableBag()
+    {
+        return this;
+    }
```

Comment:
Overrides like this don't hurt, but aren't really necessary, since withAll() has the same optimization.

```java
    @Override
    public <T> ImmutableBag<T> withAll(Iterable<? extends T> items)
    {
        if (items instanceof ImmutableBag<?>)
        {
            return (ImmutableBag<T>) items;
        }
        if (items instanceof Bag<?>)
        {
            Bag<T> bag = (Bag<T>) items;
            if (bag.isEmpty())
            {
                return this.with();
            }
            if (bag.size() == 1)
            {
                return this.with(bag.getFirst());
            }
            if (bag.sizeDistinct() < ImmutableArrayBag.MAXIMUM_USEFUL_ARRAY_BAG_SIZE)
            {
                return ImmutableArrayBag.copyFrom(bag);
            }
            return new ImmutableHashBag<>(bag);
        }
        return this.of((T[]) Iterate.toArray(items));
    }
```

Reply:
@motlin I prefer the clarity of the direct return.